src/XMakeBuildEngine/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs

```diff
@@ -390,12 +393,19 @@ private int LaunchNode(string msbuildLocation, string commandLineArgs)
 
             BackendNativeMethods.PROCESS_INFORMATION processInfo = new BackendNativeMethods.PROCESS_INFORMATION();
 
-            string appName = msbuildLocation;
+
+#if RUNTIME_TYPE_NETCORE
+            string coreRunName = NativeMethodsShared.IsUnixLike ? "CoreRun" : "CoreRun.exe";
```

Comment:
I think this should be `corerun` on non-Windows (at least according to https://github.com/dotnet/buildtools/blob/c578e891af4dea18b8f7e2524780bd4a9a3f8ce4/src/Microsoft.DotNet.Build.Tasks/PackageFiles/toolruntime.targets#L12)